src/imageResponse.js

```diff
@@ -119,69 +104,52 @@ export default {
+      log('debug', `HEAD request for ${imageDescription} can be served`);
       return;
     }
-    log('debug', `Request for ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)})`);  //eslint-disable-line max-len
+    log('debug', `Request for ${imageDescription}`);
 
     const cacheValue = await cache.getFromCache(params);
     if (cacheValue) {
-      log('info', `Cache hit for ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)})`);  //eslint-disable-line max-len
+      log('debug', `Cache hit for ${imageDescription}`);
       redirectToCachedEntity(cacheValue, params, response);
       return;
     }
 
     // Image is present but not in the correct setting
-    log('info', `Cache miss for ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)})`);  //eslint-disable-line max-len
+    log('info', `Cache miss for ${imageDescription}`);
     sendFoundHeaders(params, response);
 
     const clientStartTime = new Date();
     const browserImage = await image.magic(imagePath(params.name), params, response);
-    browserImage.stream(params.type, (err, stdout, stderr) => {
-      logErrIfNeeded(stderr, 'Live image creation stderr');
-      if (err) {
-        response.status(500).end();
-        log('error', `Error occurred while creating live image: ${err}`);
-        return;
-      }
-      const errors = [];
-      const r = stdout.pipe(response);
-      r.once('finish', () => {
-        if (errors.length === 0) {
-          log('info', `Creating live image took ${new Date() - clientStartTime}ms: ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)})`);  //eslint-disable-line max-len
-        } else {
-          log('warn', `Got an error while creating live image. Took ${new Date() - clientStartTime}ms: ${params.name}.${params.type} (${params.width}x${params.height}px, fit: ${params.fit}, blur: ${Boolean(params.blur)}). Errors: ${JSON.stringify(errors)}`);  //eslint-disable-line max-len
-        }
-        response.end();
-      });
-      r.once('error', (error) => {
-        log('error', `The live image stream hit an error: ${error}`);
-        stdout.unpipe(response);
-        errors.push(error);
-        response.end();
-        r.end();
-      });
+    browserImage.toBuffer(params.type, (err, stream) => {
       if (request) {
         request.once('close', () => {
-          log('warn', 'Client disconnected prematurely. Terminating stream');
-          stdout.unpipe(response);
-          errors.push('Client disconnected prematurely.');
+          log('warn', `Client disconnected prematurely. Terminating stream for ${imageDescription}`);
           response.end();
-          r.end();
         });
       }
 
+      if (err) {
```

Comment:
- I think the `request.once` listener should be attached *before* calling `toBuffer`. With the current implementation it'll likely only be called after we already try to send the response.
- In addition, I don't think we should even try to send the response when the request is closed, something which currently still happens.